benchmarks/src/main/scala/zio/stm/TReentrantLockBenchmark.scala

```diff
@@ -0,0 +1,104 @@
+package zio.stm
+
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.StampedLock
+
+import scala.util.Random
+
+import org.openjdk.jmh.annotations._
+
+import zio.IOBenchmarks._
+import zio.{ Managed, _ }
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Measurement(iterations = 5, timeUnit = TimeUnit.SECONDS, time = 5)
+@Warmup(iterations = 5, timeUnit = TimeUnit.SECONDS, time = 5)
+@Fork(1)
+@Threads(1)
+class TReentrantLockBenchmark {
+
+  @Param(Array("50", "100", "500"))
+  var numReaders: Int = _
+
+  @Param(Array("1", "10", "25", "50", "100"))
+  var numWriters: Int = _
+
+  @Param(Array("100"))
+  var ops: Int = _
+
+  @Param(Array("1000"))
+  var dataSize: Int               = _
+  private var data: Map[Int, Int] = _
+
+  private var zioLock: UIO[TReentrantLock] = _
+
+  private var javaLock: UIO[JavaStampedLock] = _
+
+  val rnd = new Random(0)
+
+  @Setup(Level.Trial)
+  def setup(): Unit = {
+    data = (0 to dataSize).toList.zipWithIndex.toMap
+    zioLock = TReentrantLock.make.commit
+    javaLock = UIO(new JavaStampedLock(new StampedLock()))
+  }
+
+  @TearDown(Level.Trial)
+  def tearDown(): Unit =
+    data.foreach {
+      case (k, v) => assert(k == v)
+    }
+
+  @Benchmark
+  def reentrantLock(): Unit = {
```

Comment:
This causes an error due to discarded value, please change the return value to `Int`.

Reply:
I fixed it.